genequery-rest/src/test/kotlin/gq/rest/api/OverlapControllerTest.kt

```diff
@@ -0,0 +1,164 @@
+        requestForm.genes = queryGenes
+        requestForm.speciesFrom = "hs"
+        requestForm.speciesTo = "hs"
+        requestForm.moduleName = "GSE1000_GPL96#0"
+        makeRequest(requestForm)
+                .andExpect(status().isOk)
+                .andExpect(jsonPath("$.success", equalTo(true)))
+                .andExpect(jsonPath("$.result.identifiedGeneFormat", equalTo("entrez")))
+                .andExpect(jsonPath("$.result.geneConversionMap.42", equalTo(42)))
+                .andExpect(jsonPath("$.result.overlapGenes", hasSize<Int>(0)))
+    }
+
+    @Test
+    fun testAmbiguousGenes() {
+        val requestForm = OverlapController.OverlapRequestForm()
+        val queryGenes = listOf("440915", "Abc")
+        requestForm.genes = queryGenes
+        requestForm.speciesFrom = "hs"
+        requestForm.speciesTo = "hs"
+        requestForm.moduleName = "GSE1000_GPL96#0"
+
+        makeRequest(requestForm)
+                .andExpect(status().is4xxClientError)
+                .andExpect(jsonPath("$.success", equalTo(false)))
+                .andExpect(jsonPath("$.result", nullValue()))
+                .andExpect(jsonPath("$.errors", hasSize<Int>(1)))
+                .andExpect(jsonPath("$.errors[0]", allOf(containsString("440915"), containsString("Abc"))))
+    }
+
+    @Test
+    fun testMissingModule() {
+        val requestForm = OverlapController.OverlapRequestForm()
+        val queryGenes = listOf("440915", "127703")
+        requestForm.genes = queryGenes
+        requestForm.speciesFrom = "hs"
+        requestForm.speciesTo = "hs"
+        requestForm.moduleName = "GSE1337_GPL42#0"
+
+        makeRequest(requestForm)
+                .andExpect(status().is4xxClientError)
+                .andExpect(jsonPath("$.success", equalTo(false)))
+                .andExpect(jsonPath("$.result", nullValue()))
+                .andExpect(jsonPath("$.errors", hasSize<Int>(1)))
+                .andExpect(jsonPath("$.errors[0]", allOf(containsString("Module"), containsString("not found"))))
+    }
+
+    @Test
+    fun testFewErrors() {
+        val requestForm = OverlapController.OverlapRequestForm()
+        requestForm.genes = emptyList()
+        requestForm.speciesTo = "hss"
+
+        makeRequest(requestForm)
+                .andExpect(status().is4xxClientError)
+                .andExpect(jsonPath("$.success", equalTo(false)))
+                .andExpect(jsonPath("$.result", nullValue()))
+                .andExpect(jsonPath("$.errors", hasSize<Int>(4)))
+                .andExpect(jsonPath("$.errors", hasItem(containsString("moduleName"))))
+                .andExpect(jsonPath("$.errors", hasItem(containsString("hss"))))
+                .andExpect(jsonPath("$.errors", hasItem(containsString("speciesFrom"))))
+                .andExpect(jsonPath("$.errors", hasItem(containsString("genes"))))
+    }
+
+    private fun makeRequest(form: OverlapController.OverlapRequestForm): ResultActions {
```

Comment:
Method `makeRequest` and `json` duplicate code from [GeneSetEnrichmentControllerTest](https://github.com/ctlab/genequery-kotlin/blob/master/genequery-rest/src/test/kotlin/gq/rest/api/GeneSetEnrichmentControllerTest.kt). Could you extract it to somewhere else (e.g. superclass)?